cypress/e2e/allowance.cy.ts

```diff
@@ -0,0 +1,77 @@
+import { transactionData } from '../support/constants';
+
+const fillInputAmount = (amount: string) => {
+  cy.fillInput('@amount', amount);
+};
+
+const fillInputAddress = (address: string) => {
+  cy.fillInput('@targetAddress', address);
+};
+
+const selectToken = (tokenId: string, token: string) => {
+  cy.changeSelect('@selectToken', tokenId, token);
+};
+
+const clickApproveButton = () => {
+  cy.clickButton('@approveButton');
+};
```

Comment:
duplicated, they're already in `commands.ts` file. I would recommend adapt them to follow this logic

Reply:
Want to abstract the test id to avoid typos.

I also tried to define each inside `beforeEach`, but the only way I found to type the context is to redeclare the `mocha` module, so types will be global, not wanted.

Any pattern you recommend?

``` jsx
const inputAmount = '@amount';
...

// or

const elements = {
  inputAmount: '@amount',
  ...
}
```

Reply:
 > Want to abstract the test id to avoid typos.
> I also tried to define each inside beforeEach,

It shouldn't be necessary, if there is any typo the test will fail and the CI/CD will prevent that pr from continuing if there is a failure

Reply:
I understand you.
Not specifically related to this test, but isn't it better to have defined all the elements I'm going to test?

Speaking of DX, I found 2 possible improvements:
- What if something changes in the component? Should I search and replace all references?
A commit of several lines instead of just one.
- If there are dozens of elements, should I remember all the selectors / aliases?
I can call an existing element that is not the right one and it still works.

I'm writing keeping in mind what I see on Bridd.gg. Can easily get lost:
``` jsx
.getDataTest('token-search-input').get('input[placeholder="Search name or paste address"]')
```

Sorry guys if this conversation is unnecessary.
I'm sharing with you what I felt while reviewing e2e tests and looking for opinions.
CC @0xLaruku @0xtiti

Reply:
data-test* atributes are not likely to change, it's the most robust locator. But it's true what you mention, and that is usually solve implementing page object model design pattern. In the other hand adding POM in small projects could add an unncessary layer of abstraction and complexity.

Of course there're situations when data-test* atributtes can't be added (like those inputs) and there's a risk.

Reply:
Thx Laruku! Didn't know about POM, interesting but right, completely unnecessary here

Reply:
https://github.com/0xShishigami/feo-web3-react-boilerplate/pull/6/commits/f9882f145c7df37628625e01dec0391ad6dd9e81